<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
        "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
	
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
<meta http-equiv="content-type" content="text/html;charset=utf-8" />
<link rel="stylesheet" href="./css/core.css" type="text/css" />
<link rel="stylesheet" href="./css/print.css" type="text/css" media="print" />
<title>CyFluxViz</title>
</head>

<body>

<div id="menu">
	<div id="navigation">
	
	<h1>&nbsp;&nbsp;<img alt="CyFluxViz logo" src="./images/CyFluxViz_Logo.png"></img>&nbsp;&nbsp;CyFluxViz</h1>
	<a class="menu" href="#Introduction">Introduction</a><br />
	<a class="menu" href="#Screenshots">Screenshots</a><br />
	<a class="menu" href="#Installation">Installation</a><br />
	<a class="menu" href="#Tutorial">Tutorial</a><br />
	<a href="https://sourceforge.net/projects/fluxvizplugin/files/latest/download?source=navbar" target="_blank"><img alt="CyFluxViz logo" src="./images/downloadbutton.png"></img></a>
	</div>

	<div id="news">
	<p class="news">
	<b>Download Additional Files</b><br />
	<a href="https://sourceforge.net/projects/fluxvizplugin/files/CyFluxViz-v0.93/CyFluxViz-v0.93_examples.zip/download" target="_blank">&nbsp;<img src="./images/examples.png">CyFluxViz Examples</a></br>	
	<a href="http://sourceforge.net/projects/fluxvizplugin/files/CyFluxVizToolbox-v0.03/CyFluxVizToolbox-v0.03.zip/download" target="_blank">&nbsp;<img src="./images/matlabicon.png">CyFluxVizToolbox</a></br>		
	</p>
	
	<p class="news">
	<b>Documents</b><br />
	<a href="./paper/Koenig2010_FluxViz.pdf" target="_blank">&nbsp;<img src="./images/pdf.jpeg">CyFluxViz Paper</a></br>	
	<a href="./presentation/Koenig2010_CyFluxViz_Introduction.pdf" target="_blank">&nbsp;<img src="./images/pdf.jpeg">CyFluxViz Presentation</a></br>		
	</p>
	
	<p class="news">
	<b>Release Information</b><br />
	[2013-08-07] <b><i>CyFluxViz-v0.93</i> </b><br />
	<ul>
	<li>bugfix attribute subnetworks</li>
	<li>genome-scale examples</li>
	</ul>
	</p>
	
	<p class="news">		
	[2013-08-07] <b><i>CyFluxVizToolbox-v0.03</i> </b><br />
	<ul>
	<li>bugfixes & JSON support</li>
	<li>genome-scale examples</li>
	</ul>
	</p>
	
	<p class="news">
	[2013-08-01] <b><i>CyFluxViz-v0.92</i> </b><br />
	<ul>
	<li>Mayor release for Cytoscape 2.8.3</li>
	<li>multiple bugfixes</li>
	<li>redesign GUI</li>
	<li>reimplemented FluxDistributions</li>
	<li>extended import/export for CyFluxVizToolbox</li>
	<li>multiple visual styles</li>
	</ul>
	</p>
	
	<p class="news">
	[2013-08-01] <b><i>CyFluxVizToolbox-v0.02</i> </b><br />
	<ul>
	<li>First release of COBRA support</li>
	</ul>
	</p>
	
	<p class="news">
	<b>Contact</b><br />
	<a href="http://sourceforge.net/projects/fluxvizplugin/" target="_blank">http://sourceforge.net/projects/fluxvizplugin/</a></br> 
	<a href="http://www.charite.de/sysbio/people/koenig/" target="_blank">http://www.charite.de/sysbio/people/koenig/</a><br></br>
	</p>
	<h1>&nbsp;</h1>
	</div>
	
</div>

<div id="main">

<!-- Introduction -->
<h2><a name="top"></a><a name="Introduction"></a>Introduction</h2>
<h3>CyFluxViz</h3>
<p>	
	CyFluxViz is an open-source <a href="http://www.cytoscape.org/" target="_blank">Cytoscape</a> plugin for the visualization of flux distributions in networks. CyFluxViz currently supports Cytoscape 2.8 and is available for download from <a href="http://sourceforge.net/projects/fluxvizplugin/" target="_blank">SourceForge</a>.<br /> 
</p>

<h3>Features</h3>
<p>
<ul>
<li>Import of networks (SBML, GML, XGMML, SIF, BioPAX, PSI-MI)</li>
<li>Import/export of flux distributions in a variety of formats</li>
<li>Sub-networks based on flux carrying reactions (flux subetwork) or arbitrary
network attributes</li>
<li>Flexible mapping architecture for all visual attributes</li>
<li>Supports multiple visual styles</li>
<li>Full integration with <a href="http://www.charite.de/sysbio/people/koenig/software/cysbml/" target="_blank">CySBML</a></li>
<li>Export of views in variety of formats (SVG, EPS, PDF, BMP, PNG)</li>
<li>Batch capabilities</li>
<li>Support of COBRA for Matlab via CyFluxVizToolbox</li>
</ul>
<p>
	Please try and have fun</br>
	The CyFluxViz team.
</p>
<p>
<iframe width="560" height="315" src="//www.youtube.com/embed/BBGiFU_gEAo" frameborder="0" allowfullscreen></iframe>
</p>

<h3>CyFluxVizToolbox & COBRA support</h3>
<p>	CyFluxViz supports the visualization of FBA results from <a href="http://opencobra.sourceforge.net/openCOBRA/Welcome.html" target="_blank">COBRA</a> via the Matlab CyFluxVizToolbox. The CyFluxVizToolbox, a collection of Matlab scripts, generating the CyFluxViz files from 
the COBRA FBA solutions. The CyFluxVizToolbox is available for download <a href="http://sourceforge.net/projects/fluxvizplugin/files/CyFluxVizToolbox-v0.03/CyFluxVizToolbox-v0.03.zip/download">here</a>. 
</p>
<p>
<iframe width="560" height="315" src="//www.youtube.com/embed/Ne0uLME_QN4" frameborder="0" allowfullscreen></iframe> 
</p>

<h3>Visualization of Tracer Experiments</h3>
<p>
CyFluxViz allows the visualization of tracer experiments.
</p>
<iframe width="560" height="315" src="//www.youtube.com/embed/mR_Ruv_H8Hk" frameborder="0" allowfullscreen></iframe>

<h3>Citation</h3>
<p>
	CyFluxViz is developed by <a href="http://www.charite.de/sysbio/people/koenig/">Matthias K&ouml;nig</a> and was funded by the Virtual Liver Network.
</p>
<p>
	<a href="http://www.virtual-liver.de"><img src="./images/VLN_logo.png" alt="VLN Logo" border=0 height=37 width=100></img></a>&nbsp;&nbsp;
	<a href=\"http://www.charite.de\"><img src="http://www.charite.de/sysbio/Charite.small.png" alt="Charite Logo" border=0 height=37 width=101></img></a>&nbsp;&nbsp;
</p>
Please cite the following publication when using CyFluxViz:
<table>
<tr>
	<td>
	Matthias K&ouml;nig and Hermann-Georg Holzh&uuml;tter<br />	
	<i>FluxViz - Cytoscape Plug-in for Vizualisation of Flux Distributions in Networks</i><br />
 	Genome Informatics 2010, Vol.24, p.96-103 [<a href="http://www.ncbi.nlm.nih.gov/pubmed?term=22081592" target="_blank">PubMed</a>]
 	</td>
 	<td><a href="./paper/FluxViz_Koenig2010.pdf"><img src="../../images/pdf.jpeg"></td>
</tr>
</table>
</p>


<h3>Use cases</h3>
<p>
CyFluxViz was used for the visualization and analysis of FBA results in a genome-scale network of Human hepatocyte metabolism
<table>
<tr>
	<td>
	Gille C, B&ouml;lling C, Hoppe A, Bulik S, Hoffmann S, H&uuml;bner K, Karlst&auml;dt A, Ganeshan R, K&ouml;nig M, Rother K, Weidlich M, Behre J, Holzh&uuml;tter HG. (2010) <i>HepatoNet1: a comprehensive metabolic reconstruction of the human hepatocyte for the analysis of liver physiology.</i> Mol Syst Biol., <b>6</b>:411. [<a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=pubmed&cmd=Retrieve&dopt=AbstractPlus&list_uids=20823849" target="_blank">PubMed</a>]
 	</td>
</tr>
</table>
</p>

<!-- Screenshots -->
<h2></a><a name="Screenshots"></a>Screenshots</h2>
<span class="toplink">[<a href="#top">top</a>]</span><br /><br />
<table>
	<tr>
		<td>
<a href="images/CyFluxViz-v0.86_screenshot.png"><img src="images/CyFluxViz-v0.86_screenshot.png" width="700" alt="CyFluxViz example image"/></a><br/></td>
</tr>

<tr><td>
<span class="figure_caption"><b>Fig.1</b>: CyFluxViz visualization of flux distribution in demo network and human erythrocyte network.
FBA Simulations were performed with <a href="http://www.bioinformatics.org/fasimu/" target="_blank">FASIMU</a>. Reversible reactions are
shown in green, irreversible reactions in red. CyFluxViz integrates seamlessly with 
<a href="http://www.charite.de/sysbio/people/koenig/software/cysbml/" target="_blank">CySBML</a><br/> providing semantic information for
the SBML models. 
(Click for larger version)</span></td>
</tr>
</table>

<br />
<table>
	<tr>
		<td><a href="images/fluxviz_test.png"><img src="images/fluxviz_test.png" width="300" alt="CyFluxViz example image"/></a></td>
		<td><span class="figure_caption"><b>Fig.2</b>: CyFluxViz visualization of flux distribution in human hepatocyte network. FBA Simulations were performed with <a href="http://www.bioinformatics.org/fasimu/" target="_blank">FASIMU</a>. The ATP production was set as target flux under varying oxygen conditions. For the layout a standard Cytoscape Layout algorithm was used.<br />
CyFluxViz Panel on the left with information about the selected flux distribution and selection area for the loaded flux distributions.<br /> 
(Click for larger version)</span></td>
</tr>	
<tr>
<td>
<a href="./images/standard_layout_example.png"><img alt="Standard hepatocyte network with layout example flux" src="./images/standard_layout_example.png" width="300"></img></a>
</td><td>
<span class="figure_caption"><b>Fig.3</b>: CyFluxViz visualization of <a href="http://www.bioinformatics.org/fasimu/">FASIMU</a> FBA simulation in reconstructed hepatocyte network. Metabolic network consists of glycolysis, gluconeogenesis, pentose phosphate pathway and citrate cycle. Identical network to Fig.1, but different layout algorithm.<br />
Manual generated layout was used for the visualisation. On the left the CyFluxViz Control Panel with help dialogue, flux distribution list and settings panel.
<br />
(Click for larger version)
</span>
</td>
</tr>
</table>

<!-- Installation -->
<h2><a name="Installation"></a>Installation & Startup</h2>
<span class="toplink">[<a href="#top">top</a>]</span><br/>
<h3>Install CyFluxViz</h3>
<p>
<a href="http://www.cytoscape.org/" target="_blank">Cytoscape 2.8</a> or later must be installed prior to CyFluxViz.
</p>
<p>
CyFluxViz works optimal with networks loaded in SBML format via <a href="http://www.charite.de/sysbio/people/koenig/software/cysbml/" target="_blank">CySBML</a>. We recommend to install CySBML prior to the installation of CyFluxViz and provide networks as SBML files to Cytoscape.
<p>
[1] Download the latest CyFluxViz version at <a href="http://sourceforge.net/projects/fluxvizplugin/files/latest/download" target="_blank">http://sourceforge.net/projects/fluxvizplugin/</a>.</br>
</p>
<p>
[2] Move the <span class="code">CyFluxViz-vX.XX.jar</span> into the Cytoscape plugin directory located in the Cytoscape installation directory under plugins<br />
<span class="code">$CYTO_INSTALL/plugins/</span>.</br> 
On Windows Systems <span class="code">Program Files/Cytoscape_v2.8.3/plugins/</span> on Mac 
<span class="code">/Applications/Cytoscape_v2.8.3/plugins</span>.<br/>
</p>
<h3>Update CyFluxViz</h3>
<p>
To update to current CyFluxViz version replace the old version of <span class="code">CyFluxViz-vX.X.jar</span> with the new version, to uninstall remove the <span class="code">CyFluxViz-vX.X.jar</span> from the plugin folder.
<h3>Start CyFluxViz</h3>
<p>
The plugin is loaded and installed during the next Cytoscape startup. To start CyFluxViz click the CyFluxViz icon in the Cytoscape toolbar&nbsp;<img alt="CyFluxViz logo" src="./images/CyFluxViz_Logo.png" height=20></img> which will load the CyFluxViz panel.</br>

</p>

<!-- Tutorial -->
<h2><a name="Tutorial"></a>Short Tutorial</h2>
<span class="toplink">[<a href="#top">top</a>]<br /></span>
<p>
CyFluxViz is integrated into Cytoscape at startup and available via the CyFluxViz Panel in the Cytoscape Control Panel after clicking the CyFluxViz icon in the Cytoscape toolbar&nbsp;<img alt="CyFluxViz logo" src="./images/CyFluxViz_Logo.png" height=20></img>.
</p>

<table>
<tr>
<td><a href="./images/fluxviz_panel.png" target="_blank"><img src="./images/fluxviz_panel.png" alt="CyFluxViz Control Panel" height="400"></img></a>
</td>

<td class="figure_caption">
	<b>Fig.4:</b> CyFluxViz Panel (on the left) consisting of three main areas:
<ul>
<li><b>Information Area</b> (top)</li>
<li><b>Flux Distribution Area</b> (middle)</li>
<li><b>Settings, Import and Export Area</b> (bottom)</li>
</ul>

The <b>Information Area</b> is used for the visualisation of help information or information about the selected flux distribution. By clicking the 'help button' the help menu is loaded in the Information Area.<br /> 
In the example on the left additional information for the flux distribution '03_atp_cyto.val' is displayed in the Information Area.<br /><br />
The <b>Flux Distribution Area</b> displays the loaded Flux distributions and is mainly used for the selection of the flux distribution for visualisation and image export.<br /><br />
The <b>Settings Area</b> hosts the main functionality. Additional flux distributions or simulation information can be loaded, the view can be limited to the flux containing subnetwork or flux distributions can be exported as images.
The mapping function for the flux mapping can also be adapted in this area.
(Click for larger image)
</td>
</tr>
</table>

<p>
The CyFluxViz workflow consists of (1) loading a network, (2) loading corresponding flux distributions via val files for the network and (3) selecting the flux distribution of interest for visualization.
</p>
<h3>Loading network</h3>
<p>
Network for the visualisation has to be loaded (session file or SBML import of network via CySBML).<br />
The 'sbml type' node attribute with values 'species' or 'reaction' is necessary for all nodes for CyFluxViz. This attribute is automatically generated if the network is imported as SBML via CySBML. If the 'sbml type' node attribute is not available it has to be generated manually.
</p>
<h3>Loading flux distributions</h3>
<p>
After loading the network for the flux visualisation the flux distributions for the network have to be loaded. This is managed via <b>Load val</b>. Node attributes for the val files and the necessary edge attributes (edge_val &amp; edge_val_dir) are generated.
 Flux files (<span class="code">*.val</span>) consist of id -> flux mappings corresponding to the loaded network. The used network and flux files have to be consistent in regard to the node identifiers.<br />
</p>
<p>
The loaded flux distributions are shown in the Flux Distributions Area. To visualise the fluxes select one of the loaded flux files in the table. The information about the selected flux distribution is displayed in the information panel and the flux mapping is updated based on the selected flux distribution.
</p>

<h3>Visualisation of flux subnetworks</h3>
<p>
CyFluxViz can visualise the flux containing subnetwork in the full network graph. Only the edges with (flux != 0.0) and the adjacent 'reaction' and 'species' nodes are shown. To limit the view to the flux subnetwork activate flux subnetwork in the Settings Area.
</p>

<h3>Apply additional mappings</h3>
<p>
Additional mappings of node and edge attributes can be used in the visualisation. These additional mappings are generated in the VisualMapper in the Control panel. So for example the localisation of the metabolic network elements can be mapped to the node color or gene expression values to the node size.
</p>
</div>

</body>
</html>
