<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
        "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
	
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
<meta http-equiv="content-type" content="text/html;charset=utf-8" />
<link rel="stylesheet" href="./css/core.css" type="text/css" />
<link rel="stylesheet" href="./css/print.css" type="text/css" media="print" />
<title>FluxViz</title>
</head>

<body>

<div id="menu">
<div id="navigation">
<!-- <h1>FluxViz</h1> -->
<img alt="FluxViz logo" src="./images/fluxviz_logo.png" width="170"></img><br />
<a class="menu" href="#Introduction">Introduction</a><br />
<a class="menu" href="#Installation">Installation</a><br />
<a class="menu" href="#Usage">Usage Guide</a><br />
<a class="menu" href="#Examples">Examples</a><br />
<a class="menu" href="#Resources">Additional Resources</a><br />
<a class="menu" href="#Development">Development</a><br />
<a class="menu" href="#FAQ">FAQ</a><br />
<a class="menu" href="#Bugs">Bugs</a><br />
<a class="menu" href="#Features">Comming Features</a><br />
</div>

<div id="news">
<!-- <h3 class="news">FluxViz v0.14 released</h3>  -->
<p class="news">
<a href="http://sourceforge.net/projects/fluxvizplugin/files/fluxviz-v0.14/FluxViz-v0.14.jar/download" target="_blank"><img alt="Download current version" src="./images/download_button.png" width="170"></img></a>
</p>

<p class="news">
[07|05|2010]<br />
New FluxViz version released. Additional mapping and
subnet features and full support for Cytoscape 2.7.
</p>

<p class="news">
[10|01|2010]<br />
New FluxViz version with installer. Many small buxfixes since v0.11. Additional examples
have been added.
</p>

<p class="news">
<b>Contact</b><br />
http://sourceforge.net/projects/fluxvizplugin/<br></br>
matthiaskoenig[at]users.sourceforge.net<br></br> 
http://www.charite.de/sysbio/people/koenig/<br></br>
</p>


</div>


</div>


<div id="main">
<!-- Introduction -->
<h2><a name="top"></a><a name="Introduction"></a>Introduction</h2>
<p>
<b>Motivation</b><br></br>
Methods like FBA and kinetic modeling are widely used to calculate fluxes in metabolic networks. For the analysis and understanding of simulation results and experimentally measured fluxes visualization software within the network context is indispensable.<br></br>
</p>
<p>
<b>Results</b><br></br>
We present <span class="fluxviz">FluxViz</span>, an open-source Cytoscape [1] plug-in for the visualization of flux distributions in molecular interaction networks. 
<ul>
<li>import of networks (SBML, GML, XGMML, SIF, BioPAX, PSI-MI)</li>
<li>import of flux distributions in a variety of formats</li>
<li>subnetworks based on flux carrying reactions (flux subetwork) or arbitrary</li>
network attributes
<li>flexible mapping architecture for all visual attributes</li>
<li>export of views in variety of formats (SVG, EPS, PDF, BMP, PNG)</li>
<li>batch capabilities</li>
</ul>
FluxViz was used for the visualization and analysis of the metabolic network of
the human hepatocyte [2].
</p>
<p>
<b>Availability</b><br></br>
The platform-independent program is an open-source project, freely available at http://sourceforge.net/projects/fluxvizplugin/ under GNU public license, including manual, tutorial and examples.
</p>
<p> Publications originating from the use of <span class="fluxviz">FluxViz</span> should include citations of the relevant publications (see
	below).:
<table>
<tr>
<td>Matthias K&ouml;nig and Hermann-Georg Holzh&uuml;tter (2010)<br></br>
<i>FluxViz - Cytoscape Plug-in for Vizualisation of Flux Distributions in Networks</i><br></br>
 Genome Informatics [accepted] </td><td><br></br><a href="./paper/FluxViz_Koenig2010.pdf"><img src="./images/pdf.jpeg">
 </td></tr>
 </table>
 <br></br>

</p>
<table>
	<tr>
		<td><a href="images/fluxviz_test.png"><img src="images/fluxviz_test.png" width="300" alt="FluxViz example image"/></a></td>
		<td><span class="figure_caption"><b>Fig.1</b>: FluxViz visualization of flux distribution in human hepatocyte network. FBA Simulations were performed in <a href="http://www.bioinformatics.org/fasimu/">FASIMU</a>. The ATP production was set as target flux under varying oxygen conditions. For the layout a standard Cytoscape Layout algorithm was used.<br />
FluxViz Panel on the left with information about the selected flux distribution and selection area for the loaded flux distributions.<br /> 
(Click for larger version)</span></td>
</tr>	

<tr>
<td>
<a href="./images/standard_layout_example.png"><img alt="Standard hepatocyte network with layout example flux" src="./images/standard_layout_example.png" width="300"></img></a>
</td><td>
<span class="figure_caption"><b>Fig.2</b>: FluxViz visualization of <a href="http://www.bioinformatics.org/fasimu/">FASIMU</a> FBA simulation in reconstructed hepatocyte network. Metabolic network consists of glycolysis, gluconeogenesis, pentose phosphate pathway and citrate cycle. Identical network to Fig.1, but different layout algorithm.<br />
Manual generated layout was used for the visualisation. On the left the FluxViz Control Panel with the FluxViz help system, the flux distribution area and the settings area.
<br />
(Click for larger version)
</span>
</td>
</tr>
</table>

<p>
<a href="http://www.cytoscape.org/">Cytoscape</a> is an open source bioinformatics software platform for visualizing molecular interaction networks and integrating these interactions with gene expression profiles and other state data.
<span class="fluxviz">FluxViz</span> is an Cytoscape Plugin which enables the visualization of flux distributions in molecular interaction networks. One usage case are FBA simulations in metabolic networks.<br />
Edge and node attributes of the network view are changed according to the selected flux distribution (s. <b>Fig.1</b> and <b>Fig.2</b>). 
<span class="fluxviz">FluxViz</span> is not limited to biochemical reaction networks but can be applied for the visualisation of arbitrary fluxes in arbitrary graphs.
</p>

<p>
<span class="fluxviz">FluxViz</span> was primarily developed as tool for the visualisation of fluxes in metabolic networks and as frontend for <a href="http://www.bioinformatics.org/fasimu/">FASIMU</a>, a flexible software for flux-balance computation series in large metabolic networks.
</p>

<p>
<table>
<tr>
[1] Killcoyne S., Carter G.W. Smith J., Boyle J., <i>Cytoscape: a community-based
framework for network modeling.</i>, Methods Mol Biol, 563: 219 - 239, 2009<br></br>
</tr>
<tr>
[2] Gille, B&ouml;lling, Hoppe, Bulik, Hoffmann, H&uuml;bner, Karlst&auml;dt, Ganeshan, K&ouml;nig, Rother,
Weidlich, Behre, Holzh&uuml;tter, <i>A metabolic network of the human hepatocyte to simulate
liver functions</i>, MSB, accepted 2010.
</tr>
</table>
</p>


<h3>Features <span class="fluxviz">FluxViz v0.14</span></h3>
<ul>
<li>Import flux distributions
	<ul>
		<li>Import FASIMU val files</li>
		<li>Import flux distributions as Cytoscape attributes</li>
	</ul>
</li>
<li>Mapping of flux distributions on network attributes
	<ul>
		<li>Edge size, edge direction and edge tooltip are used for the visualisation of the flux through the corresponding edge</li>
		<li>Additional node and edge attributes can be used to represent additional data like for example localisation as node color or gene expression as node size.</li>
		<li>Flexible mapping functions</li>
	</ul>
</li>
<li>Cycle through loaded flux distributions
	<ul>
		<li>On the fly generation of flux distribution mapping with consistent layout</li>
	</ul>
</li>
<li>Annotation of simulation information to flux distributions
	<ul>
		<li>import of FASIMU simulations file</li>
	</ul>
</li>

<li>Limit visualization to subnetwork
	<ul>
		<li>flux containing subnetworks (only flux containing edges and nodes)</li>
		<li>subnetworks based on arbitrary attributes</li>
	</ul>
	

</li>

<li>Additional mappings possible (like gene expression or localisation)</li>

<li>Image export of generated flux distributions
	<ul>
		<li>export of images in all supported Cytoscape formats (PDF, EPS, PNG, SVG, BMP)</li>
	</ul>
</li>
</ul>
<p>
<span class="fluxviz">FluxViz</span> is developed and maintained by <a href="http://www.charite.de/sysbio/people/koenig/"> Matthias Koenig</a> at the <a href="http://www.charite.de/sysbio/">Computational Systems Biochemistry</a> group of the Charite Berlin.
</p>
<p>
Project homepage<br />
<a href="http://www.charite.de/sysbio/people/koenig/software/fluxviz/help/">http://www.charite.de/sysbio/people/koenig/software/fluxviz/help/</a><br />
Project hosted at<br />
<a href="http://sourceforge.net/projects/fluxvizplugin/">http://sourceforge.net/projects/fluxvizplugin/</a>.
</p>

<!-- Installation -->
<h2><a name="Installation"></a>Installation</h2>
<span class="toplink">[<a href="#top">top</a>]</span><br />
<p>
Cytoscape has to be installed prior to the plugin installation. The plugin works with Cytoscape versions 2.6 and 2.7. <span class="fluxviz">FluxViz</span> can be installed manually or via the Cytoscape Plugin Manager.
</p>

<h3>Manual Installation</h3>
<p>
<span class="fluxviz">FluxViz</span> is avaiblable for download at <a href="http://sourceforge.net/projects/fluxvizplugin/">http://sourceforge.net/projects/fluxvizplugin/</a>. 
The current version can also be downloaded via the download link on the left. 
</p>
<p>
Download the latest Plugin version and move <span class="code">fluxviz-vx.x.jar</span> into the Cytoscape plugin directory located in the Cytoscape installation directory under plugins<br /> <span class="code">$CYTO_INSTALL/plugins/</span>.<br />
The plugin is loaded and installed during the next Cytoscape start.
</p>
<p>
To update to a newer version of <span class="fluxviz">FluxViz</span> replace the old <span class="code">fluxviz-vx.x.jar</span> with the current version. Furthermore remove the FluxViz files for the old installation under <span class="code">$HOME/.cytoscape/2.x/plugins/</span>
</p>

<h3>Installation via the Cytoscape Plugin Manager</h3>
<p>
	<span class="fluxviz">FluxViz</span> is also available via the Cytoscape plugin manager.
	The current version in the Cytoscape Plugin Manager can be older than the current available version for manual installation.
</p>

<p>
The Cytoscape Plugin Manager is available in the Cytoscape menu<br />
<span class="code">Plugins -> Manage Plugins</span>.<br />
The <span class="fluxviz">FluxViz</span> Plugin is listed in 'Functional Enrichment'. Select the Plugin and click install. 
</p>
<p>
To update an older version select<br />
<span class="code">Plugins -> Update Plugins</span>.<br />
</p>
<table>
	<tr>
		<td><a href="./images/plugin_manager.png"><img src="./images/plugin_manager.png" width="300px" alt="Cytoscape Plugin Manager"></img></a></td>
		<td><span class="figure_caption"><b>Fig.2</b>: FluxViz installation via the Cytoscape Plugin Manager.<br /> (Click for larger version)</span></td>
	</tr>
</table>


<!-- Usage guide for FluxViz -->
<h2><a name="Usage"></a>Usage guide</h2>
<span class="toplink">[<a href="#top">top</a>]<br /></span>
<p>
<span class="fluxviz">FluxViz</span> is integrated into Cytoscape at startup and available via the <span class="fluxviz">FluxViz</span> Panel in the Cytoscape Control Panel.<br />
</p>

<table>
<tr>
<td><a href="./images/fluxviz_panel.png" target="_blank"><img src="./images/fluxviz_panel.png" alt="FluxViz Control Panel" height="400"></img></a>
</td>

<td class="figure_caption">
	<b>Fig.3:</b> <span class="fluxviz">FluxViz</span> Panel (on the left).
The <span class="fluxviz">FluxViz</span> Panel consists of three areas:

<ul>
<li><b>Information Area</b> on top called Information</li>
<li><b>Flux Distribution Area</b> in the middle called Flux distributions</li>
<li><b>Settings, Import and Export Area</b> on the bottom called Settings</li>
</ul>

The <b>Information Area</b> is used for the visualisation of help information or information about the selected flux distribution. By clicking the 'help button' the help menu is loaded in the Information Area.<br /> 
In the example on the left additional information for the flux distribution '03_atp_cyto.val' is displayed in the Information Area.<br /><br />
The <b>Flux Distribution Area</b> displays the loaded Flux distributions and is mainly used for the selection of the flux distribution for visualisation and image export.<br /><br />
The <b>Settings Area</b> hosts the main functionality. Additional flux distributions or simulation information can be loaded, the view can be limited to the flux containing subnetwork or flux distributions can be exported as images.
The mapping function for the flux mapping can also be adapted in this area.
(Click for larger image)
</td>
</tr>
</table>
	
<h3>Requirements for <span class="fluxviz">FluxViz</span></h3>
<p>
To visualise flux distributions<br /> 
<b>1.</b> Network for the visualisation has to be loaded (session file or SBML import of network).<br />
Session files are loaded via<br />
<span class="code">File -> Open</span><br />
SBML files should always be imported into a new empty session<br />
<span class="code">File -> New -> Session</span><br />
<span class="code">File -> Import -> Network (multiple file types)</span><br />
For the SBML import the <a href="http://www.charite.de/sysbio/people/koenig/software/sbmlreader2/help/">SBMLReader2</a> should be used.
</p>

<p>
<b>2.</b> Flux files (<span class="code">*.val</span>) consisting of id -> flux mappings corresponding to the loaded network have to be imported. The used network and the loaded flux files have to be consistent.<br />
The val files can be imported by clicking on the <b>Load val</b> button and selecting the val files.
</p>

<p>
<b>3.</b> 'sbml type' node attribute with the classification 'species' and 'reaction' for all nodes is necessary. This attribute is automatically generated if the network is imported as SBML. If the 'sbml type' node attribute is not available it has to be generated manually (s. <a href="#FAQ">FAQ</a>). Normally this attribute should be available after the import of the network.
</p>

<p>
The normal workflow is to load a network and the corresponding val files into Cytoscape. Afterwards additional information from the FASIMU <span class="code">simulations</span> file can be loaded via the <b>Load sim</b> button.<br />
By selecting the different flux distributions in the Flux distributions area the mappings are applied to the current network.
</p>

<h3>Main FluxViz Functionality</h3>
<h4>Loading network (import SBML)</h4>
The network is imported via <br />
<span class="code">File -> Import -> Network (multiple file types)</span>.<br />
For the SBML import the <a href="the SBMLReader2">SBMLReader2</a> Plugin for Cytoscape should be used. This reader supports all SBML levels and versions and reads the edge attribute from the file.<br />
If the network has been imported before also the Cytoscape session file <span class="code">*.cys</span> can be used.

<h4>Loading flux distributions</h4>
<p>
After loading the network for the flux visualisation the flux distributions for the network have to be loaded. This is managed via <b>Load val</b>. Node attributes for the val files and the necessary edge attributes (edge_val &amp; edge_val_dir) are generated.
</p>
<p>
The available flux distributions are shown in the Flux Distributions Area in the <span class="fluxviz">FluxViz</span> Control Panel. To visualise the fluxes select one of the loaded flux files in the table. The information about the selected flux distribution is displayed in the information panel and the flux mapping is updated based on the selected flux distribution.
</p>

<h4>Deleting flux distributions</h4>
<p>
To delete flux distributions (val attributes) the node val attributes have to be deleted. The flux val files are updated automatically. Don't delete the *.val_edge or *.val_edge_dir files without the corresponding node *.val files.
</p>

<h4>Visualisation of flux subnetworks</h4>
<p>
FluxViz can visualise the flux containing subnetwork in the full network graph. Only the edges with (flux != 0.0) and the adjacent 'reaction' and 'species' nodes are shown. To limit the view to the flux subnetwork activate flux subnetwork in the Settings Area.
</p>

<h4>Apply additional mappings</h4>
<p>
Additional mappings of node and edge attributes can be used in the visualisation. These additional mappings are generated in the VisualMapper in the Control panel. So for example the localisation of the metabolic network elements can be mapped to the node color or gene expression values to the node size.
</p>

<h4>Reuse of layouts</h4>
<p>
For the understanding of fluxes in metabolic networks a consistent layout of the metabolic networks is necessary. All the Cytoscape layout algorithms can be used or manually created layouts be used for the visualisation of the fluxes. The Layouts can be stored by storing the Cytoscape session. Additional flux simulations can be mapped on the same network.
</p>



<!-- examples -->
<h2><a name="Examples"></a>Examples</h2>
<span class="toplink">[<a href="#top">top</a>]<br /></span>
<p>
Example networks with corresponding flux data are available in <span class="fluxviz">FluxViz</span>.
The examples are part of the FluxViz.jar and are installed in<br />
<span class="code">$HOME/.cytoscape/2.6/plugins/FluxVizData/examples/</span>.
</p>
<p>
The examples consist of 
</p>
<table>
	<tr><td><span class="code">network.sbml</span></td><td>SBML of the network</td></tr>
	<tr><td><span class="code">/val/*.val</span></td><td>val files of FASIMU FBA simulations</td></tr>
	<tr><td><span class="code">simulations</span></td><td>FASIMU simulation description file</td></tr>
	<tr><td><span class="code">network.cys</span></td><td>Cytoscape session file of network</td></tr>
</table>
<p>
The examples can be manually loaded from Cytoscape by loading the <span class="code">*.cys</span> Session file or importing the <span class="code">*.sbml</span> SBML network. The layout information is lost during the SBML import.</p>
<p>
Cytoscape can be started with the FluxViz example networks by setting the <span class="code">-P FluxVizExample</span> flag during Cytoscape startup.
</p>
<table>
	<tr><th>Network</th><th>Cytoscape flag</th> <th></th></tr>
	<tr><td>Standard Hepatocyte network</td><td><span class="code">-P FluxVizExample=1</span></td><td></td></tr>
	<tr><td>Erthrocyte network with layout</td><td><span class="code">-P FluxVizExample=2</span></td><td></td></tr>
	<tr><td>Standard Hepatocyte network with layout</td><td><span class="code">-P FluxVizExample=3</span></td><td></td></tr>
	<tr><td>HepatoCyte Core network</td><td><span class="code">-P FluxVizExample=4</span></td><td></td></tr>
</table>
<table>
<tr>
<td>
<a href="./images/standard_example.png"><img alt="Standard hepatocyte network example flux." src="./images/standard_example.png" width="200"></img></a>
</td><td>
<span class="figure_caption"><b>Fig.6</b>: Hepatocte example flux of FASIMU FBA simulation. Automated network layout algorithm used.<br />
<span class="code">-P FluxVizExample=1</span><br />
(Click for larger version)</span>
</td>
</tr>
	
<tr>
<td>
<a href="./images/erythrocyte_example.png"><img alt="Erythrocyte example flux" src="./images/erythrocyte_example.png" width="200"></img></a>
</td><td>
<span class="figure_caption"><b>Fig.7</b>: Erythrocyte example flux of FASIMU FBA simulation. Metabolic network with glycolysis and pentose phosphate pathway.<br />
<span class="code">-P FluxVizExample=2</span><br />
(Click for larger version)
</span>
</td>
</tr>

<tr>
<td>
<a href="./images/standard_layout_example.png"><img alt="Standard hepatocyte network with layout example flux" src="./images/standard_layout_example.png" width="200"></img></a>
</td><td>
<span class="figure_caption"><b>Fig.8</b>: Hepatocyte example flux of FASIMU FBA simulation. Metabolic network consisting of glycolysis, gluconeogenesis, pentose phosphate pathway and citrate cycle.<br />
<span class="code">-P FluxVizExample=3</span>
<br />
(Click for larger version)</span>
</td>
</tr>
</table>

<!--
<h2>Implementation Information</h2>
Attribute files:
For the visualisation of a flux mode three different attribute files are necessary.
1. *.val: the node attribute val files which stores the fluxes for the nodes (reactions, transporters). 
2. *.val_edge: the edge attribute file which stores the fluxes for the edges adjacent to nodes with flux.
3. *.val_edge_dir: edge attribute file which stores the direction of the fluxes for the edges adjacent to nodes with flux.
The edge attribute files 2. and 3. are generated from the *.val node attribute file with the help of the 'sbml type' attribute.
Therefore an attribute 'sbml type' is necessary for the plugin to function. 
#TODO: creation of a 'sbml type' attribute (classification of the nodes in species and reactions).
-->

<!-- Resources -->
<h2><a name="Resources"></a>Additional Resources</h2>
<span class="toplink">[<a href="#top">top</a>]<br /></span>
Use the following link for the citation of FluxViz:
<table align="top" cellpadding="5">
<td>2010</td><td>Matthias König and Hermann-Georg Holzhütter, <i>FluxViz - Cytoscape Plug-in for Vizualisation of Flux Distributions in Networks</i>, Genome Informatics, (2010) [submitted]</td><td><a href="./paper/FluxViz_Koenig2010.pdf">PDF</a></td></tr>
</table>
<ul>
	<li><a href="../doc/index.html">API documentation</a></li>
	<li>Additional help information is available via the 'help' button in the Settings Area in the <span class="fluxviz">FluxViz</span> Control panel.</li>
<li>short <span class="fluxviz">FluxViz</span> presentation <a href="./presentation/fluxviz_introduction.pdf">fluxviz_introduction.pdf</a></li>
</ul>

<h3>Links</h3>
<p>
FluxViz<br />
<a href="http://www.charite.de/sysbio/people/koenig/software/fluxviz/help/index.html">http://www.charite.de/sysbio/people/koenig/software/fluxviz/help/index.html</a><br /> 
<a href="http://sourceforge.net/projects/fluxvizplugin/">http://sourceforge.net/projects/fluxvizplugin/</a><br />  
<a href="http://www.charite.de/sysbio/people/koenig/">http://www.charite.de/sysbio/people/koenig/</a>
</p>
<p>
SBMLReader2<br />
<a href="http://www.charite.de/sysbio/people/koenig/software/sbmlreader2/help/">http://www.charite.de/sysbio/people/koenig/software/sbmlreader2/help/</a>
<a href="http://sourceforge.net/projects/sbmlreader2/">http://sourceforge.net/projects/sbmlreader2/</a>
</p>
<p>
Cytoscape<br /> 
<a href="http://www.cytoscape.org/">http://www.cytoscape.org/</a>
</p>
<p>
Cytoscape plugins<br /> 
<a href="http://chianti.ucsd.edu/cyto_web/plugins/index.php">http://chianti.ucsd.edu/cyto_web/plugins/index.php</a>
</p>
<p>
FASIMU<br />
<a href="http://www.bioinformatics.org/fasimu/">http://www.bioinformatics.org/fasimu/</a>
</p>

<!-- Development -->
<h2><a name="Development"></a>Development</h2>
<span class="toplink">[<a href="#top">top</a>]<br /></span>
<h3>Subversion Access</h3>
<p>
The project is open source. Everybody can participate.
</p>
<p>
This project's SourceForge.net Subversion repository can be checked out through SVN with the following instruction set:<br />
<span class="code">svn co https://fluxvizplugin.svn.sourceforge.net/svnroot/fluxvizplugin fluxvizplugin </span> 
</p>

<!-- FAQ -->
<h2><a name="FAQ"></a>FAQ</h2>
<span class="toplink">[<a href="#top">top</a>]<br /></span>
<h4>The links to the Help Tutorial, JavaDoc and Contact are not working.</h4>
<p>
	The <span class="code">defaultWebBrowswer</span> has to be set under<br />
	<span class="code">Edit -> Preferences -> Properties</span><br />
	and the proxy has to be set correctly under <br />
	<span class="code">Edit -> Preferences -> ProxyServer</span>
</p>
<h4>I get the error message no complete node attribute 'sbml type' when I try to import val files.</h4>
<p>
	For <span class="fluxviz">FluxViz</span> to work with <span class="code">*.val</span> files a node attribute 'sbml type' is necessary. Normally this is generated during the SBML import of the network. If the network was imported in any other format than SBML this attribute is missing and has to be manually generated.<br />
Select the 'Node Attribute Browser' and click <span class="code">Create new attribute -> String attribute</span>. Enter 'sbml type' as attribute name.<br />
Now all nodes have to be classified in 'reaction' and 'species'. Select all reaction nodes and open the 'Attribute Batch Editor'. Select <span class="code">Set - sbml type</span> to 'reaction. Select all compound nodes and open the 'Attribute Batch Editor'. Select <span class="code">Set - sbml type</span> to 'species'.
</p>
<p>
Now the import of the val files should work.	
</p>


<!-- Bugs -->
<h2><a name="Bugs"></a>Bugs in <span class="fluxviz">FluxViz-v0.1</span></h2>
<span class="toplink">[<a href="#top">top</a>]</span><br />
<ul>
<li><span class="fluxviz">FluxViz</span> has problems with multiple networks in v.0.1. Therefor only a single network with the corresponding fluxes should be loaded in one session.</li>
</ul>

<!-- Features -->
<h2><a name="Features"></a>Comming Features in <span class="fluxviz">v0.2</span></h2>
<span class="toplink">[<a href="#top">top</a>]</span><br />
<ul>
<li>support for multiple networks in one session</li>
<li>export of images in all supported Cytoscape formats (PDF, EPS, PNG, SVG, BMP)</li>
<li>alternative input formats (to support additional edge/flux data)</li>
<li>speed improvments in the flux subnetwork generation and visual mapping</li>
<li>improved javadoc documentation</li>
</ul>

</div>

</body>
</html>
